packages/manager/apps/cloud/client/app/dbaas/logs/detail/kibana/add/logs-kibana-add.controller.js

```diff
@@ -0,0 +1,142 @@
+class LogsKibanaAddModalCtrl {
+  constructor(
+    $q,
+    $stateParams,
+    $uibModalInstance,
+    $translate,
+    CucControllerHelper,
+    kibanaInfo,
+    LogsKibanaService,
+    LogsConstants,
+  ) {
+    this.$stateParams = $stateParams;
+    this.$q = $q;
+    this.CucControllerHelper = CucControllerHelper;
+    this.kibanaInfo = kibanaInfo;
+    this.LogsKibanaService = LogsKibanaService;
+    this.LogsConstants = LogsConstants;
+    this.$uibModalInstance = $uibModalInstance;
+    this.$translate = $translate;
+    this.serviceName = $stateParams.serviceName;
+    this.kibana = this.LogsKibanaService.getNewKibana();
+    this.kibanaPrice = { price: '' };
+    this.initLoaders();
+  }
+
+  $onInit() {
+    this.isEdit = this.constructor.checkIsEdit(this.kibanaInfo);
+    if (this.isEdit) {
+      this.populateKibana();
+    } else {
+      this.clearKibana();
+      this.accountDetails
+        .load()
+        .then(() => {
+          this.ovhSubsidiary = this.accountDetails.data.me.ovhSubsidiary;
+          return this.$q.all([this.catalog.load()]);
+        })
+        .then(() => {
+          const selectedCatalog = this.catalog.data.plans.find(
+            (plan) => plan.planCode === this.LogsConstants.LDP_PLAN_CODE,
+          );
+          const selectedFamily = selectedCatalog.addonsFamily.find(
+            (addon) => addon.family === this.LogsConstants.ADD_ON_FAMILY.NEW,
+          );
+          const kibanaCapacities = selectedFamily.addons.find(
+            (add) =>
+              add.plan.planCode ===
+              this.LogsConstants.CONSUMPTION_REFERENCE.KIBANA,
+          );
+          const kibanaUnit = kibanaCapacities.plan.details.pricings.default.find(
+            (capa) =>
+              capa.capacities.includes(this.LogsConstants.CONSUMPTION_CAPACITY),
+          );
+          this.kibanaPrice.price = kibanaUnit.price.text;
+        });
+    }
+  }
+
+  initLoaders() {
+    this.catalog = this.CucControllerHelper.request.getArrayLoader({
+      loaderFunction: () =>
+        this.LogsKibanaService.getOrderCatalog(this.ovhSubsidiary),
+    });
+    this.accountDetails = this.CucControllerHelper.request.getHashLoader({
+      loaderFunction: () =>
+        this.LogsKibanaService.getAccountDetails(this.serviceName),
+    });
+  }
+
+  showKibanaPrice() {
+    return this.$translate.instant(
+      'logs_kibana_modal_description_help',
+      {
+        t0: this.kibanaPrice.price,
+      },
+      undefined,
+      false,
+      'sceParameters', // Expose devise symbol from API without sanitization
+    );
+  }
+
+  clearKibana() {
+    this.title = 'logs_kibana_modal_add_title';
+    this.kibana.description = '';
+  }
+
+  populateKibana() {
+    this.title = 'logs_kibana_modal_edit_title';
+    this.kibana.description = this.kibanaInfo.description;
+  }
+
+  static checkIsEdit(kibanaInfo) {
+    return kibanaInfo !== undefined;
+  }
+
+  cancel() {
+    this.$uibModalInstance.dismiss();
```

Comment:
We should prefer using the `layout: modal` to have modal with states but it can require further work so it can be taken care of in another PR

Reply:
FYI...we are planning to remove the dependency on uibModel. Please consider using layout modal

Reply:
Hello,

I understand, but as say @marie-j, it will require much more work:

```
grep -r 'uibModalInstance' packages/manager/apps/cloud/client/app/dbaas/logs/ | wc -l
      58
```

Reply:
Thanks @marie-j and @radireddy for pointing out that we should avoid using `$uibModal` and prefer using the `layout` decorator instead but it implies also to review the routing part which acts more as refactor in my opinion.

I'm OK to keep the `uibModalInstance` for this Pull Request.

Removing `$uibModal` can be handled into a separate Pull Request as already did here: #2077.

Thanks!